File: DSA/Khushi_590028673/Day12/Q1.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution{
    public:

    vector<int> numberOfLines(vector<int>&widths, string s){
        int lines=1;
        int currentwidth=0;
        for(char ch:s){
            int w=widths[s[ch]-'a'];

            if(currentwidth+w<=100){
                currentwidth+=w;
            }
            else{
                lines++;
                currentwidth=w;
            }
        }

        return {lines, currentwidth};
    }
};


int main(){
    vector<int>widths={10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10};

    string s="abcefghijklmnopqrstuvwxyz";

    Solution obj;

    vector<int>result= obj.numberOfLines(widths, s);

    cout<<"number of lines: "<<result[0]<<endl;
    cout<<"width of last line: "<< result[1]<<endl;


    return 0;
}

File: DSA/vaishnavi_joshi_590016947/Day12/Q1.java
import java.util.*;

public class LinesToWriteString {
    public static int[] numberOfLines(int[] widths, String s) {
        int lines = 1;
        int currentWidth = 0;

        for (char ch : s.toCharArray()) {
            int width = widths[ch - 'a'];

            if (currentWidth + width > 100) {
                lines++;
                currentWidth = width;
            } else {
                currentWidth += width;
            }
        }

        return new int[]{lines, currentWidth};
    }

    public static void main(String[] args) {
        int[] widths = {
            10,10,10,10,10,10,10,10,10,10,
            10,10,10,10,10,10,10,10,10,10,
            10,10,10,10,10,10
        };

        String s = "abcdefghijklmnopqrstuvwxyz";

        System.out.println(Arrays.toString(numberOfLines(widths, s)));
    }
}